graylog2-web-interface/src/components/configurations/URLWhitelistFormModal.jsx

```diff
@@ -0,0 +1,158 @@
+// @flow
+import React from 'react';
+import PropTypes from 'prop-types';
+import uuid from 'uuid/v4';
+import connect from 'stores/connect';
+import { IfPermitted } from 'components/common';
+import { Button } from 'components/graylog';
+import BootstrapModalForm from 'components/bootstrap/BootstrapModalForm';
+import UrlWhiteListForm from 'components/configurations/UrlWhiteListForm';
+import CombinedProvider from 'injection/CombinedProvider';
+import PermissionsMixin from 'util/PermissionsMixin';
+import type { WhiteListConfig } from 'stores/configurations/ConfigurationsStore';
+
+const { CurrentUserStore } = CombinedProvider.get('CurrentUser');
+const { ConfigurationsActions, ConfigurationsStore } = CombinedProvider.get('Configurations');
+
+const URL_WHITELIST_CONFIG = 'org.graylog2.system.urlwhitelist.UrlWhitelist';
+
+type State = {
+  config: WhiteListConfig,
+  isValid: boolean
+};
+
+type Props = {
+  newUrlEntry: string,
+  onUpdate: () => void,
+  configuration: {},
+  currentUser: {permissions: Array<string>},
+  formType: string,
+};
+
+
+class URLWhitelistFormModal extends React.Component<Props, State> {
```

Comment:
Most of the code in this component should be very similar to parts of what we have in `UrlWhiteListConfig.jsx`, I think we should try and make a single component doing this, so we can reuse it better whenever is needed.